contracts/CurrencyNetwork.sol

```diff
@@ -191,26 +191,23 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Authorizable, debtTracking
     }
 
     /**
-     * @notice send `_value` token to `_to` from `msg.sender`
-     * The fees will be payed by the sender, so `_value` is the amount received by `_to`
-     * @param _to The address of the recipient
-     * @param _value The amount of token to be transferred
+     * @notice send `_value` along `_path`
+     * The fees will be payed by the sender, so `_value` is the amount received by receiver
+     * @param _value The amount to be transferred
      * @param _maxFee Maximum fee the sender wants to pay
-     * @param _path Path between msg.sender and _to
+     * @param _path Path of transfer starting with msg.sender and ending with receiver
      * @param _extraData extra data bytes to be logged in the Transfer event
      **/
     function transfer(
-        address _to,
         uint64 _value,
         uint64 _maxFee,
         address[] calldata _path,
         bytes calldata _extraData
     )
         external
     {
+        require(msg.sender == _path[0], "The path must start with msg.sender");
```

Comment:
Is it necessary to include the sender in the path? pro cons?

Reply:
Pro: it is more readable and every functions have a similar definition of the path
Cons: gas consumption

Reply:
Saves about 1600 gas to remove the sender from the path (but then we also need to reindex stuff in the loop)

Reply:
Lets leave it with the full path.

Reply:
It would create to much problems for inner functions, as we can not easy prepend an address to the path .